docs/learning/guidelines/release-tags-guidelines.md

```diff
@@ -1,66 +1,188 @@
+
+`src/internal/cross-package.ts` exports individual top-level cross-package internal APIs from `src/internal/`.
+
+`src/internal/Symbols.ts` defines [Symbol](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Symbol)s that identify nested internal APIs.
+
+When evaluating an existing or new `@internal` API, apply the following policies in order by priority.
+
+1. If the API does not need to be `@internal` - i.e., it does not meet the criteria defined [above](#choosing-appropriate-release-tags) - apply the appropriate release tag instead
+1. Convert a nested API to a top-level API, if possible (e.g., a static class method that does not access private members of the class can be converted to a top-level function).
+1. Move a top-level API into `src/internal/`
+1. A top-level, cross-package API should be exported from `src/internal/cross-package.ts`. Don't export single-package APIs from `cross-package.ts`- they can be imported wthin the package using a relative path.
+1. For a nested API that cannot be converted to a top-level API, convert its name to a Symbol using the example below as a guide.
+
+### Converting a nested API
+
+Imagine we have the following `@internal` API that we want to make inaccessible to code outside of itwinjs-core:
 
 ```ts
-/** Sample description of an exported API item.
- * @public
- * @extensions
- */
+export class Thing {
+  /** @internal */
+  public close(): void {
+    // implementation goes here
+  }
+}
+
+```
+
+Add to `src/internal/Symbols.ts` the following, if a `_close` symbol doesn't already exist (keeping the list of symbols sorted alphabetically):
+
+```ts
+export const _close = Symbol.for("close_INTERNAL_ONLY_DO_NOT_USE");
 ```
+
+Update `Thing` to rename `close` to use the symbol, and deprecate the existing `close` method:
+
+```ts
+export class Thing {
+  /** @internal */
+  public [_close]: () => void {
+    // implementation goes here
+  }
+  
+  /** @internal
+   * @deprecated in 4.8. The thing will automatically be closed when it is no longer in use. This API is for internal use only and will soon be removed.
+   */
+  public close(): void {
+    this[_close]();
+  }
+}
+```
+
+Update all callers of `close` in itwinjs-core to use `[_close]` instead.
+
+The symbols in `Symbols.ts` are not exported from `core-backend.ts`, so they are inaccessible to packages outside of itwinjs-core. We preserve the original `@internal` API to avoid introducing a breaking API change for external callers (who should not be using it in the first place). The deprecation tag will warn them to fix their code to remove their dependency on the internal API. If we were reasonably certain no one was depending on the `close` method, we could simply delete it.
+
+### Transition plan
+
+In iTwin.js 5.0, we will be able to begin enforcing the internal API policy without regard for breaking changes.
```

Comment:
@aruniverse @calebmshafer @wgoehrig can we commit to the listed changes for iTwin.js 5.0?

Reply:
I'm ok with this as long as we give fair/advanced warning to App teams and they know what they're getting themselves into.

Changes below, like standardizing on ESM is a dream that I'm not 100% sure we'll be able to realize in time.

Reply:
> Changes below, like standardizing on ESM 

@wgoehrig suggested standardizing on ESM was in scope. Is that the only one you potentially object to? (You said "like" that one). If we can't commit to any of them, we're not going to solve the internal API (and hence the API stability) problem.

Reply:
> Changes below, like standardizing on ESM is a dream that I'm not 100% sure we'll be able to realize in time.

Let's add it to the 5.0 list and see where we land with the impact and effort needed.

@pmconne I'm good with all of the changes suggested here other than a few smaller comments I made.

I'd like @grigasp and @GerardasB to comment as well.